src/Sylius/Behat/Service/Converter/IriConverter.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Service\Converter;
+
+use ApiPlatform\Core\Api\IdentifiersExtractor;
+use ApiPlatform\Core\Api\IdentifiersExtractorInterface;
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Api\OperationType;
+use ApiPlatform\Core\Api\ResourceClassResolverInterface;
+use ApiPlatform\Core\Api\UrlGeneratorInterface;
+use ApiPlatform\Core\Bridge\Symfony\Routing\RouteNameResolverInterface;
+use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
+use ApiPlatform\Core\DataProvider\OperationDataProviderTrait;
+use ApiPlatform\Core\DataProvider\SubresourceDataProviderInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use ApiPlatform\Core\Exception\InvalidIdentifierException;
+use ApiPlatform\Core\Exception\ItemNotFoundException;
+use ApiPlatform\Core\Exception\RuntimeException;
+use ApiPlatform\Core\Identifier\CompositeIdentifierParser;
+use ApiPlatform\Core\Identifier\IdentifierConverterInterface;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Util\AttributesExtractor;
+use ApiPlatform\Core\Util\ResourceClassInfoTrait;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Routing\Exception\ExceptionInterface as RoutingExceptionInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * Copied from src/Bridge/Symfony/Routing/IriConverter.php to add getIriFromItemInSection method
+ */
+final class IriConverter implements IriConverterInterface
```

Comment:
This is obviously a hot-fix 🖖 The problem was, after recent changes in Symfony (between versions 5 and 6), there were some services that were not reset before and are now, and one of them is the token storage. That means, that our `PathPrefixProvider` which was *stateful* before (as it uses the `UserContext` to resolve the prefix) does not work [on some occasions](https://github.com/Sylius/Sylius/pull/14376/files#diff-649a0eb931d4880a02e494ed534d1b956a49cf22f191382522150db425c7d9f0R1196) (e.g. does not provide an admin user). The hot-fix which I propose **for now,** is to make it stateless by passing the desired section (admin/shop) in which we're resolving the Behat scenario. The truth is **we always now** in which section we are in Behat steps, so this resolver could be stateless for Behat from the very beginning :dancer: 
I had to copy the whole `IriConverter` class to add a new method, I will work on a better solution (or at least cleaner one), but it can count as refactor in Behat which can be done even after 1.12-beta 🖖